.github/workflows/validate.yml

```diff
@@ -0,0 +1,49 @@
+name: validate
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
+
+jobs:
+  linters:
+    permissions:
+      contents: read      # for actions/checkout to fetch code
+      pull-requests: read # for golangci/golangci-lint-action to fetch pull requests
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        go-version: [1.19.x]
+        # FIXME(thaJeztah): enable once build-tags are fixed.
+        # os: [ubuntu-22.04, windows-2019]
+        os: [ubuntu-22.04]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.50.1
```

Comment:
Do we need to specify the version? I'd rather just inherit the latest tagged version unless there's a specific reason not to... not a big deal if something starts breaking to add a temp pin then...

Reply:
Good one; so I have run into issues if it's "too old" (not supporting older Go versions), and most repositories I work on I prefer to willingly update (not getting caught by surprise that things start failing).

Given that this repository is a lot less active, perhaps using "latest" would be good, so that we don't forget to update to newer versions etc.

I'll double check what the correct way is to use "latest", and update